apps/web/src/app/_components/reviews/new-role-dialogue.tsx

```diff
@@ -0,0 +1,147 @@
+
+  const mutation = api.role.create.useMutation({
+    onSuccess: () => {
+      setIsSuccess(true);
+      setRoleName(form.getValues().title);
+
+      setTimeout(() => {
+        window.location.reload();
+      }, 1000);
+    },
+    onError: (error) => {
+      toast({
+        title: "Submission Error",
+        description: error.message || "Something went wrong. Please try again.",
+        variant: "destructive",
+      });
+    },
+  });
+
+  async function onSubmit(values: z.infer<RoleRequestType>) {
+    try {
+      await mutation.mutateAsync(values);
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  return (
+    <Dialog>
+      <DialogTrigger asChild>
+        <Button className="h-9 rounded-full border-none border-cooper-yellow-500 bg-cooper-yellow-500 px-4 py-3 text-sm font-semibold text-white hover:border-cooper-yellow-300 hover:bg-cooper-yellow-300">
+          + Create New Role
+        </Button>
+      </DialogTrigger>
+
+      <DialogContent className="w-full bg-white">
+        {isSuccess ? (
+          <div className="flex flex-col items-center justify-center gap-4 py-8">
+            <h2 className="text-xl font-semibold text-cooper-green-500">
+              Success!
+            </h2>
+            <p className="text-center text-gray-600">
+              The {roleName} co-op has been successfully added to{" "}
+              {company.data?.name}.
+            </p>
+          </div>
+        ) : (
+          <>
+            <DialogHeader>
+              <DialogTitle>Create New Role</DialogTitle>
+              <DialogDescription>
+                Request a new role for {company.data?.name}
+              </DialogDescription>
+            </DialogHeader>
+
+            <Form {...form}>
+              <div className="flex flex-col gap-4 py-4">
+                <FormField
+                  control={form.control}
+                  name="title"
+                  render={({ field }) => (
+                    <FormItem>
+                      <Label>Role Name</Label>
+                      <FormControl>
+                        <Input type="string" variant="dialogue" {...field} />
+                      </FormControl>
+                    </FormItem>
+                  )}
+                />
+                <FormField
+                  control={form.control}
+                  name="description"
+                  render={({ field }) => (
+                    <FormItem>
+                      <Label>Description</Label>
+                      <FormControl>
+                        <Textarea variant="dialogue" {...field} />
+                      </FormControl>
+                    </FormItem>
+                  )}
+                />
+              </div>
+            </Form>
+            <DialogFooter>
+              <Button
+                type="submit"
+                className="border-none bg-cooper-blue-400 text-white hover:bg-cooper-blue-600"
+                onClick={form.handleSubmit(onSubmit)}
```

Comment:
Let's create a submit handler so we can validate the input data in there. For example, here is part of `next` function in the review form which performs the zod form validation:

```
const next = async (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
 e.preventDefault();
 const fields = steps[currentStep - 1]?.fields;
 const output = await form.trigger(fields as FieldName[], {
    shouldFocus: true,
 });

 if (!output) {
   return;
 }
```

Because right now I can enter nothing for both of the fields and it works which shouldn't be the case

Reply:
Also for this file it would look specifically like:
'''
  async function onSubmit(values: z.infer<RoleRequestType>) {
    try {
      const result = await form.trigger(["title", "description"], {
        shouldFocus: true,
      });

      if (!result) {
        return;
      }

      await mutation.mutateAsync(values);
    } catch (error) {
      console.error(error);
    }
  }
'''

Reply:
But we also need to add '<FormMessage className="text-sm" />' underneath each of the FormControl tags for the error message to work